rapid7/metasploit-framework
# Add a binary safe `write_file` method for powershell session type

## Summary
This PR add `powershell` support for `write_file` method. This method is binary safe and compresses the file on the host to reduce network traffic. It divides the data into smaller chunks, compresses them and send on the target system.

## Verification Steps
Transfer and verify checksum of few binary and non-binary files.
```
>> file = open('/home/pingport80/linux_shell_stage.elf','rb')
=> #<File:/home/pingport80/linux_shell_stage.elf>
>> _write_file_powershell('check.bin',file.read)
=> "194"
```
